src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -936,6 +974,26 @@ private void SetSessionHandleOptions(SafeWinHttpHandle sessionHandle)
             SetSessionHandleTlsOptions(sessionHandle);
             SetSessionHandleTimeoutOptions(sessionHandle);
             SetDisableHttp2StreamQueue(sessionHandle);
+            SetTcpKeepalive(sessionHandle);
+        }
+
+        private unsafe void SetTcpKeepalive(SafeWinHttpHandle sessionHandle)
+        {
+            if (_tcpKeepAliveEnabled)
+            {
+                var tcpKeepalive = new Interop.WinHttp.tcp_keepalive
+                {
+                    onoff = 1,
+                    keepaliveinterval = (uint)_tcpKeepAliveInterval.TotalMilliseconds,
+                    keepalivetime = (uint)_tcpKeepAliveTime.TotalMilliseconds
```

Comment:
I think you need something checking if the time spans are `Timeout.InfiniteTimeSpan`, or these will get set to `(uint)-1`.

Reply:
This will cast to `uint.MaxValue` milliseconds (~50 days), which might be an appropriate setting for "infinite" here. I would prefer to not special case this. Added a comment and a unit test.